stl/src/cthread.cpp

```diff
@@ -114,10 +114,14 @@ _CRTIMP2_PURE _Thrd_result __cdecl _Thrd_create(_Thrd_t* thr, _Thrd_start_t func
     _Thrd_result res;
     _Thrd_binder b;
     int started = 0;
-    _Cnd_t cond;
-    _Mtx_t mtx;
-    _Cnd_init(&cond);
-    _Mtx_init(&mtx, _Mtx_plain);
+
+    std::_Aligned_storage_t<_Cnd_internal_imp_size, _Cnd_internal_imp_alignment> cond_storage;
+    _Mtx_internal_imp_t mtx_storage{};

```

Comment:
Is it ok to make cond and mtx adjacent here?

Reply:
Do you mean false sharing?
It is currently avoided by oversize of `_Cnd_internal_imp_size`, at least for 64 bits.
In general, I think, you can avoid it by `alignas(hardware_constructive_interference_size)`, but:
 * Not sure if it ever worth for `mutex` and `condition_variable` used in conjunction as they are used by the same waiting thread, and then the same signaling thread.
 * Not sure if it worth for such heavy operation as thread start.

Reply:
Yes. I have only vague awareness of this type of problem.
(As to the align cost if there are, I think it will not be significant, as I've already replaced `calloc` versions with in-place versions.)

Reply:
I've force-pushed this pr to avoid such changes.